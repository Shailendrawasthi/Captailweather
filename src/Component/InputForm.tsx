import React, { FC, useState } from 'react';
import { Container, Box, Typography, TextField, Button } from '@mui/material';
import axios from 'axios';
import { useNavigate } from 'react-router-dom';

const InputForm: React.FC = () => {
    const navigate =useNavigate();

    const [Data, setData] = useState<any>( );
    const [input, setInput] = useState<string>("");


    const handleChange=async()=>{
        const response =await axios.get(`https://restcountries.com/v3.1/name/${input}`);
        setData(response);
        console.log(response.data);
        navigate('/Details',{state:{data:response.data}});
    }

    return (
        <Container maxWidth="xs"
            sx={{ marginTop: '100px', height: '400px', border: '1px solid black', bgcolor: "rgb(233, 224, 224)" }}>

            <Box style={{ display: 'flex', flexDirection: 'column', alignItems: 'center', marginTop: "50px" }}>
                <Typography variant='h4'>Country Data</Typography>
                <TextField  style={{marginBottom:'10px'}}   id='textfield' label='Entry Country'
                onChange={(e) => setInput(e.target.value)} />
                <Button
                    style={{ padding: 10, paddingRight: '80px', paddingLeft: '80px' }}
                    variant="contained"
                    onClick={handleChange} >Submit</Button>
            </Box>

        </Container>
    );
}
export default InputForm;