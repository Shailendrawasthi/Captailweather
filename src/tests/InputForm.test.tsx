import { shallow } from "enzyme";
import InputForm from "../Component/InputForm";

jest.mock('react-router-dom', () => ({
    ...jest.requireActual('react-router-dom') as any,
    useNavigate: () => ({
      navigate: jest.fn().mockImplementation(() => ({})),
    }),
}));

describe('testing inputform component', () => {
    
    let wrapper: any;

    beforeEach(() => {
        wrapper = shallow(<InputForm />);
    });

    it('renders snapshot for InputForm component', () => {
        expect(wrapper).toMatchSnapshot();
    });

    it('renders non-empty component without crashes', () => {
        expect(wrapper.exists()).toBe(true);
    });
    
    it('should render a <div />', () => {
        const result = wrapper.find('div').length;
        expect(result).toEqual(0);
    });

    it('checking Submit button', () => {
        expect(wrapper.containsMatchingElement('Submit')).toBeTruthy();
    });


    it(' render a <TextField />', () => {
        const result = wrapper.find('#textfield').length;
        expect(result).toEqual(1);
    });
    
});