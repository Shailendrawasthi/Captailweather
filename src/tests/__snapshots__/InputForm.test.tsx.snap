// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`testing Home component renders snapshot for InputForm component 1`] = `ShallowWrapper {}`;

exports[`testing inputform component renders snapshot for InputForm component 1`] = `ShallowWrapper {}`;
